From dae160a5024598ef29d588c0163236b973ec0604 Mon Sep 17 00:00:00 2001
From: Richard Greer <rg@braneaudio.com>
Date: Tue, 17 Oct 2023 11:31:57 -0500
Subject: [PATCH] bootz: add hab support

---
 cmd/bootz.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cmd/bootz.c b/cmd/bootz.c
index 4f024bde5f..78c21d4740 100644
--- a/cmd/bootz.c
+++ b/cmd/bootz.c
@@ -57,6 +57,14 @@ static int bootz_start(struct cmd_tbl *cmdtp, int flag, int argc,
 	if (bootm_find_images(flag, argc, argv, images->ep, zi_end - zi_start))
 		return 1;
 
+#ifdef CONFIG_IMX_HAB
+	extern int authenticate_image(
+			uint32_t ddr_start, uint32_t raw_image_size);
+	if (authenticate_image(images->ep, zi_end - zi_start) != 0) {
+		printf("Authenticate zImage Fail, Please check\n");
+		return 1;
+	}
+#endif
 	return 0;
 }
 
-- 
2.34.1

