From 52d7a7ac0a7fc9be36fed1f0b18e036ea6914c0c Mon Sep 17 00:00:00 2001
From: Richard Greer <rg@braneaudio.com>
Date: Tue, 17 Oct 2023 11:34:00 -0500
Subject: [PATCH] hab: fix overflow

---
 arch/arm/mach-imx/hab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/hab.c b/arch/arm/mach-imx/hab.c
index 55317abba2..068b076b11 100644
--- a/arch/arm/mach-imx/hab.c
+++ b/arch/arm/mach-imx/hab.c
@@ -446,7 +446,7 @@ static void display_event(uint8_t *event_data, size_t bytes)
 static int get_hab_status(void)
 {
 	uint32_t index = 0; /* Loop index */
-	uint8_t event_data[128]; /* Event data buffer */
+	uint8_t event_data[sizeof(struct record)]; /* Event data buffer */
 	size_t bytes = sizeof(event_data); /* Event size in bytes */
 	enum hab_config config = 0;
 	enum hab_state state = 0;
-- 
2.34.1

